### PR TITLE
feat(codemods): add codemod for updating icons packages

### LINKED ISSUE #8516: Create codemod for migrating teams to new icons API
As part of #8139, we updated the signature of our icon components to now use a `size` prop. This issue is for creating a codemod to automatically update most examples of usage through a codemod. This should include updating the following usage patterns:

```jsx
import React from 'react';
import { Add32, Bee24 as bee, Caret20, DownArrow16 } from '@carbon/icons-react';

const types = {
  add: Add32,
  bee: Bee24,
  ref,
  test: (props) => React.createElement(bee, {
    ref,
    size: 24,
    ...props,
  }),
};

function RendersIconDirectly() {
  return (
    <>
      <Add32 />
      <Bee24 />
      <Caret20 />
      <DownArrow16 />
    </>
  );
}

function RendersIconWithProps(props) {
  return (
    <>
      <Add32 aria-label="test" {...props} />
      <Bee24 aria-label="test" {...props} />
      <Caret20 aria-label="test" {...props} />
      <DownArrow16 aria-label="test" {...props} />
    </>
  );
}
```

### PR DESCRIPTION
**Pull request by joshblack**
Closes #8516 

This PR adds in the initial codemod for the Icon API updates. There will be a separate PR that will include all of the component updates. This also introduces the migration guide for `@carbon/icons-react` for v11.

#### Changelog

**New**

- Add new transform in `codemods` for updating icons-react imports

**Changed**

**Removed**

#### Testing / Reviewing

- Go into the codemods directory `cd codemods`
- Run the following command `yarn jscodeshift --parser babylon -t transforms/icons-react-size-prop.js '../packages/react/src/components'`
- Verify that files have changed with the intended API updates

### CODE REVIEW

File: docs/migration/11.x-carbon-icons-react.md [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+      <Add size={24} />
+      <Add size={20} />
+      <Add />
+    </>
+  );
+}
+```
+
+_Note: by default, the `size` prop will point to the 32x32px version of the SVG
+asset_
```
Comment (jnm2377) [inline docs/migration/11.x-carbon-icons-react.md]: I think by default it's pointing to 16x16 versions. When I ran the script, only icons size 32 and 20 were receiving "size" props, where as 16 looked like default.
Screen Shot 2021-05-03 at 9 09 23 PM
Screen Shot 2021-05-03 at 9 09 45 PM
Screen Shot 2021-05-03 at 9 10 07 PM
Screen Shot 2021-05-03 at 9 10 43 PM
Screen Shot 2021-05-03 at 9 11 20 PM

**Review by jnm2377 — APPROVED** [review]
Ran the script and the API was updated. It did return some errors for some components, but otherwise updated the rest of the files: 
Screen Shot 2021-05-03 at 9 08 06 PM

File: docs/migration/11.x-carbon-icons-react.md (hunk 1 above)
Comment (joshblack) [reply docs/migration/11.x-carbon-icons-react.md]: Yes! Great catch, I should update 👍

**Review by tay1orjones — APPROVED** [review]
Transforms look as expected 👍 

```
Results: 
0 errors
425 unmodified
0 skipped
87 ok
Time elapsed: 6.320seconds
```

I couldn't find an example of an aliased icon, but it looks like you covered that with a test.

**Pull request merged**